src/engine_leveldb.cpp

```diff
@@ -234,6 +233,52 @@ void ustore_snapshot_create(ustore_snapshot_create_t* c_ptr) {
     db.snapshots[*c.id] = level_snapshot;
 }
 
+void ustore_snapshot_export(ustore_snapshot_export_t* c_ptr) {
+    ustore_snapshot_export_t& c = *c_ptr;
+    return_error_if_m(c.db, c.error, uninitialized_state_k, "DataBase is uninitialized");
+    level_db_t& db = *reinterpret_cast<level_db_t*>(c.db);
+
+    // Find snapshot
+    leveldb::ReadOptions scan_options;
+    scan_options.fill_cache = false;
+    if (c.id) {
+        auto it = db.snapshots.find(c.id);
+        return_error_if_m(it != db.snapshots.end(), c.error, args_wrong_k, "The snapshot does'nt exist!");
+        level_snapshot_t& snap = *reinterpret_cast<level_snapshot_t*>(c.id);
+        scan_options.snapshot = snap.snapshot;
+    }
+
+    // Get iterator
+    level_iter_uptr_t it;
+    try {
+        it = level_iter_uptr_t(db.native->NewIterator(scan_options));
+    }
+    catch (...) {
+        *c.error = "Fail To Create Iterator";
+        return;
+    }
+
+    // Create new DB
+    level_native_t* native_db = nullptr;
+    level_status_t status = leveldb::DB::Open(db.options, c.path, &native_db);
+    return_error_if_m(status.ok(), c.error, args_wrong_k, "Couldn't create a new database");
+
+    // Copy all data into new DB
+    leveldb::WriteBatch batch;
+    for (it->SeekToFirst(); it->Valid(); it->Next())
```

Comment:
This will fail on any remotely interesting size... You are inserting the whole database into memory... You will run our of it.